pymongo/database.py

```diff
@@ -541,36 +541,39 @@ def command(self, command, value=1, check=True,
                                  check, allowable_errors, read_preference,
                                  codec_options, session=session, **kwargs)
 
-    def _list_collections(self, sock_info, slave_okay, criteria=None,
-                          session=None):
+    def list_collections(self, sock_info, slave_okay, filter=None,
+                         session=None, batch_size=0):
         """Internal listCollections helper."""
-        criteria = criteria or {}
-        cmd = SON([("listCollections", 1), ("cursor", {})])
-        if criteria:
-            cmd["filter"] = criteria
+        filter = filter or {}
+        cmd = SON([("listCollections", 1),
+                   ("cursor", {"batchSize": batch_size}),
+                   ("filter",filter)])
 
         if sock_info.max_wire_version > 2:
             coll = self["$cmd"]
             with self.__client._tmp_session(session, close=False) as s:
                 cursor = self._command(
                     sock_info, cmd, slave_okay, session=s)["cursor"]
                 return CommandCursor(coll, cursor, sock_info.address, session=s,
-                                     explicit_session=session is not None)
+                                     explicit_session=session is not None, batch_size=batch_size)
         else:
             coll = self["system.namespaces"]
+            if "name" in filter:
+                filter["name"] = coll.database.name + "." + filter["name"]
             res = _first_batch(sock_info, coll.database.name, coll.name,
-                               criteria, 0, slave_okay,
+                               filter, 0, slave_okay,
                                CodecOptions(), ReadPreference.PRIMARY, cmd,
-                               self.client._event_listeners, session=None)
+                               self.client._event_listeners, session=None, batch_size=batch_size)
             data = res["data"]
             cursor = {
                 "id": res["cursor_id"],
                 "firstBatch": data,
                 "ns": coll.full_name,
             }
-            return CommandCursor(coll, cursor, sock_info.address)
+            return CommandCursor(coll, cursor, sock_info.address, batch_size=batch_size)
 
-    def collection_names(self, include_system_collections=True, session=None):
+    def collection_names(self, include_system_collections=True,
+                         session=None, batch_size=0, filter=None):
```

Comment:
Let's not change this method signature just yet. The spec says:
"If a driver already has a method to perform one of the listed tasks, there is no need to change it. Do not break backwards compatibility when adding new methods."

However, you can change this method and [`Collection.options`](https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/collection.py#L1954) to use `list_collections`.